From 89ae1b1b94b750af8519f7adb1402131cc1aca42 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Thu, 23 Oct 2014 19:00:44 +0800
Subject: [PATCH 116/117] CHERRY PICK: mt9m114: Use REG_EXPO_COARSE to get
 exposure time

This patch fixes g_exposure function in mt9m114 driver
to get exposure time by reading REG_EXPO_COARSE.

Change-Id: I6b4fc86edee92b2b236eca84c227a60537be16bf
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-2846
Tracked-On: https://bz01p-vied.ir.intel.com/show_bug.cgi?id=4385
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/25209
Reviewed-by: Jing Tao <jing.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c b/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
index 4e6826a..303c908 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
@@ -1272,7 +1272,7 @@ static int mt9m114_g_exposure(struct v4l2_subdev *sd, s32 *value)
 
 	/* the fine integration time is currently not calculated */
 	ret = mt9m114_read_reg(client, MISENSOR_16BIT,
-			       MISENSOR_COARSE_INTEGRATION_TIME, &coarse);
+			       REG_EXPO_COARSE, &coarse);
 	if (ret)
 		return ret;
 
-- 
1.9.3

